processor/transformprocessor/internal/traces/condition.go

```diff
@@ -0,0 +1,56 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+type condFunc = func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) bool
```

Comment:
Should we make this a "3" stage check?
1. Check for Resource (some nodes of the decision tree may be replaced with true/false).
2. Check for InstrumentationLibrary (some nodes of the decision tree may be replaced with true/false).
3. Check for the Span/Metric/Log -> true/false

I think this way we don't have to repeat the checks on the "resource" for every span. I may have already mentioned this idea, but I think it may be an important optimization.

Reply:
you are correct, but may affect the "supported" function signatures, so we better do not open yet the API to allow external funcs until we finalize that.